src/common/envalid.ts

```diff
@@ -18,19 +18,36 @@ const cleanSpec = {
   VALUES_INPUT: str({ desc: 'The chart values.yaml file', default: undefined }),
 }
 
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-export const cleanEnvironment = () => {
-  let pEnv: any = process.env
-  const path = `${pEnv.ENV_DIR}/.secrets`
-  if (pEnv.ENV_DIR && existsSync(path)) {
-    const result = config({ path }) // this sets vars from .env onto process.env
-    if (result.error) console.error(result.error)
-    pEnv = { ...pEnv, ...result.parsed }
+export const taskEnvSpec = {
+  OTOMI_FLAGS: json({ default: {} }),
+  TEAM_IDS: json({ desc: 'A list of team ids in JSON format' }),
+}
+
+export const cleanEnvironment = (
+  spec: Record<string, any> = cliEnvSpec,
+  returnFunc = false,
+): Record<string, any> | CallableFunction => {
+  const func = (): Record<string, any> => {
+    let pEnv: any = process.env
+    // load local .env if we have it, for devs
+    let path = `${process.cwd()}/.env`
+    if (existsSync(path)) {
```

Comment:
Duplicate logic from L34 to L43 should be extracted into a function to make it more clean.